elements/lisk-p2p/src/peer/base.ts

```diff
@@ -587,4 +579,26 @@ export class Peer extends EventEmitter {
 					internalState: assignInternalInfo(peerInfo, this._peerConfig.secret),
 			  };
 	}
+
+	private _decodeNodeInfo(data: unknown): P2PNodeInfo {
+		try {
+			if (typeof data !== 'string' || !isHexString(data)) {
+				throw new Error('Invalid encoded data');
+			}
+			return codec.decode<P2PNodeInfo>(this._rpcSchemas.nodeInfo, Buffer.from(data, 'hex'));
+		} catch (error) {
+			throw new InvalidNodeInfoError((error as Error).message);
+		}
+	}
+
+	private _decodePeerInfo(data: unknown): P2PPeerInfo {
+		try {
+			if (typeof data !== 'string' || !isHexString(data)) {
+				throw new Error('Invalid encoded data');
+			}
+			return codec.decode<P2PPeerInfo>(this._rpcSchemas.peerInfo, Buffer.from(data, 'hex'));
+		} catch (error) {
+			throw new InvalidPeerInfoError((error as Error).message);
+		}
```

Comment:
Maybe we can move these functions to utils?
`decodePeerInfo(data: unknown, schema: peerInfo): P2PPeerInfo`

Also, we can use encoding peer/nodeInfo functions that encode and return encoded strings. But we can do it in some other issue